pkg/maps/ctmap/ctmap.go

```diff
@@ -54,154 +48,32 @@ const (
 	CtTypeIPv4Global
 )
 
-type CtKey interface {
-	Dump(buffer *bytes.Buffer) bool
-}
-
-type CtKey6 struct {
-	addr    types.IPv6
-	sport   uint16
-	dport   uint16
-	nexthdr u8proto.U8proto
-	flags   uint8
-}
-
-func (key CtKey6) Dump(buffer *bytes.Buffer) bool {
-	if key.nexthdr == 0 {
-		return false
-	}
-
-	if key.flags&TUPLE_F_IN != 0 {
-		buffer.WriteString(fmt.Sprintf("%s IN %s %d:%d ",
-			key.nexthdr.String(),
-			key.addr.IP().String(),
-			key.sport, key.dport),
-		)
-
-	} else {
-		buffer.WriteString(fmt.Sprintf("%s OUT %s %d:%d ",
-			key.nexthdr.String(),
-			key.addr.IP().String(),
-			key.dport,
-			key.sport),
-		)
-	}
-
-	if key.flags&TUPLE_F_RELATED != 0 {
-		buffer.WriteString("related ")
-	}
-
-	return true
-}
-
-type CtKey4 struct {
-	addr    types.IPv4
-	sport   uint16
-	dport   uint16
-	nexthdr u8proto.U8proto
-	flags   uint8
-}
-
-func (key CtKey4) Dump(buffer *bytes.Buffer) bool {
-	if key.nexthdr == 0 {
-		return false
-	}
-
-	if key.flags&TUPLE_F_IN != 0 {
-		buffer.WriteString(fmt.Sprintf("%s IN %s %d:%d ",
-			key.nexthdr.String(),
-			key.addr.IP().String(),
-			key.sport, key.dport),
-		)
-
-	} else {
-		buffer.WriteString(fmt.Sprintf("%s OUT %s %d:%d ",
-			key.nexthdr.String(),
-			key.addr.IP().String(),
-			key.dport,
-			key.sport),
-		)
-	}
-
-	if key.flags&TUPLE_F_RELATED != 0 {
-		buffer.WriteString("related ")
-	}
-
-	return true
-}
-
-type CtKey6Global struct {
-	daddr   types.IPv6
-	saddr   types.IPv6
-	sport   uint16
-	dport   uint16
-	nexthdr u8proto.U8proto
-	flags   uint8
-}
+// ServiceKey is the interface describing protocol independent key for services map.
+type ServiceKey interface {
```

Comment:
This should be renamed to `CtKey` which is the interface for the common functions of `CtKey4` and `CtKey6`